Applied-Upstream: commit: e0f5d0529ef3fd95959198c41d0529e0b54cff0a
Author: Joshua Strobl <me@joshuastrobl.com>
Last-Update: 2022-03-20
Description: [PATCH] Fix regression in IconTasklist. Fixes #79.

Behavior before: You were able to move non-pinned applications.
Behavior after 49fe42cd822fd672fc95d4eb962e0357b7f71afd: An undesired `pinned` check was introduced that prevented these apps from being moved.

This commit reverts that change as well as introduces a supplmental buttons check that will also attempt to filter out any `.desktop` from an application ID for buttons, if we cannot find the button with an app id.
---
 src/applets/icon-tasklist/IconTasklistApplet.vala | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/applets/icon-tasklist/IconTasklistApplet.vala b/src/applets/icon-tasklist/IconTasklistApplet.vala
index 20c033b7..47df6505 100644
--- a/src/applets/icon-tasklist/IconTasklistApplet.vala
+++ b/src/applets/icon-tasklist/IconTasklistApplet.vala
@@ -262,9 +262,12 @@ public class IconTasklistApplet : Budgie.Applet {
 			}
 		} else { // Doesn't start with file://
 			unowned IconButton? button = null;
+			string app_id_without_desktop_suffix = app_id.replace(".desktop", "");
 
 			if (this.buttons.contains(app_id)) { // If buttons contains this app_id
 				button = this.buttons.get(app_id);
+			} else {
+				button = this.buttons.get(app_id_without_desktop_suffix);
 			}
 
 			if (button != null) {
@@ -348,7 +351,7 @@ public class IconTasklistApplet : Budgie.Applet {
 		string first_app_id = first_app.id.to_string();
 		if (app.app_info != null) { // properly group new apps with their pinned version
 			string launcher = this.get_app_launcher(app.app_info.get_filename());
-			if (this.buttons.contains(launcher) && this.buttons.get(launcher).pinned) {
+			if (this.buttons.contains(launcher)) {
 				first_app_id = launcher;
 			}
 		}
-- 
2.34.1

